devito/equation.py

```diff
@@ -87,9 +96,26 @@ def evaluate(self):
             except AttributeError:
                 if bool(rules):
                     eq = eq.xreplace(rules)
-
         return eq
 
+    @property
+    def _flatten(self):
+        """
+        Flatten vectorial/tensorial Equation into list of scalar Equations.
+        """
+        if self.lhs.is_Matrix:
```

Comment:
Could this not cause problems if, for example, we have a scalar `lhs`? That is, could the call to this from operator pass anything without the `is_Matrix` property?

Reply:
As long as las is a devito object, and at this point should always be, then no it should be fine

Reply:
As long as lhs is a devito object, and at this point should always be, then no it should be fine